### PR TITLE
Tests and Functions

### PR DESCRIPTION
**Pull request by aalanders**

### CODE REVIEW

File: testing_exercise/testing.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   for(var i = 0; i < number; i++){
+        answer= answer.concat(array);
+   }
+   return answer;
+}
+
+
+function acceptNumbersOnly(){
+    for(var i=0; i<arguments.length; i++){
+        if(arguments[i] === NaN){
```
Comment (mmmaaatttttt) [inline testing_exercise/testing.js]: careful with this comparison: `NaN === NaN` will return `false`!

File: testing_exercise/testing.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if(arguments[i] !== Number(arguments[i])){
+            return false;
+        };
+    }
+    return true;
+}
+
+function mergeArrays(array,array2){
+        var answer = array.concat(array2)
+        return answer.sort();
```
Comment (mmmaaatttttt) [inline testing_exercise/testing.js:37]: careful with sort here (per our conversation from this morning)

File: testing_exercise/testing.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+function mergeObjects(obj, obj2) {
+    for (var key in obj2) {
+        if (obj2.hasOwnProperty(key)){
+			obj[key] = obj2[key];
```
Comment (mmmaaatttttt) [inline testing_exercise/testing.js:43]: sometimes your indentation is a little inconsistent on github -- it's possible you've got spaces and tabs interspersed in your code.

**Review by mmmaaatttttt — COMMENTED** [review]
a couple of tweaks to the test exercises would be :100:, but overall :+1:!

File: canvas_exercise/shapes_game/index.js [hunk 4]
```diff
@@ -1,36 +1,123 @@
-window.addEventListener("load", function() {
+//get random shape on screen
```
Comment (tigarcia) [inline canvas_exercise/shapes_game/index.js:1]: Looks good.  There are a few bugs you can work on.  Whenever I press the space bar to start the game, my score starts out as -1.  It seems to be related to how you attribute points in the keyup callback.  Also, After a game is over, I can't see my score.  Try showing it in the game screen after 30 seconds is up.

File: call_apply_bind_exercise/callApplyBind.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+// ## Call Apply Bind Exercises
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js:1]: Good job on this!

File: prototypes_exercise/prototypes.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
+//this is constructor function, specific to every instance. If it's not just instance, can put on prototype
+
+function Person(firstName,lastName,favoriteColor,favoriteNumber,favoriteFoods=[]){
```
Comment (tigarcia) [inline prototypes_exercise/prototypes.js:3]: Looks good here too! Give reduce a try if you have some extra time.

**Review by tigarcia — COMMENTED** [review]
Call apply and bind + the prototypes exercise looks good.  Taking a look at hacker snooze now.

File: ajax_with_jquery_exercise/hackernews2.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
+
+
+$(function(){
```
Comment (tigarcia) [inline ajax_with_jquery_exercise/hackernews2.js:3]: Good start! Try to get the favoriting working when you can.  At least the creating of a favorite.

**Pull request closed without merging**